source/InfiniteLoader/InfiniteLoader.js

```diff
@@ -0,0 +1,139 @@
+/** @flow */
+import FlexTable from '../FlexTable'
+import React, { Component, PropTypes } from 'react'
+import shouldPureComponentUpdate from 'react-pure-render/function'
+import VirtualScroll from '../VirtualScroll'
+
+/**
+ * Higher-order component that manages lazy-loading for "infinite" data.
+ * This component decorates a FlexTable or VirtualScroll and prefetches rows as a user scrolls to J.I.T. load data.
+ * It is intended as a convenience component; fork it if you'd like finer-grained control over data-loading.
+ */
+export default class InfiniteLoader extends Component {
+  shouldComponentUpdate = shouldPureComponentUpdate
+
+  static propTypes = {
+    /** Children must be either FlexTable or VirtualScroll */
+    children (props, propName, componentName) {
+      let error
+      React.Children.forEach(props.children, child => {
+        if (!(child.type === FlexTable || child.type === VirtualScroll)) {
+          error = new Error(`InfiniteLoader only accepts children of types FlexTable or VirtualScroll not ${child.type}`)
+        }
+      })
+      return error
+    },
+    /**
+     * Function responsible for tracking the loaded state of each row.
+     * It should implement the following signature: (index: number): boolean
+     */
+    isRowLoaded: PropTypes.func,
+    /**
+     * Callback to be invoked when more rows must be loaded.
+     * It should implement the following signature: ({ startIndex, stopIndex }): Promise
+     * The returned Promise should be resolved once row data has finished loading.
+     * It will be used to determine when to refresh the list with the newly-loaded data.
+     * This callback may be called multiple times in reaction to a single scroll event.
+     */
+    loadMoreRows: PropTypes.func.isRequired,
+    /**
+     * Number of rows in list; can be arbitrary high number if actual number is unknown.
+     */
+    rowsCount: PropTypes.number,
+    /**
+     * Threshold at which to pre-fetch data.
+     * A threshold X means that data will start loading when a user scrolls within X rows.
+     * This value defaults to 10.
```

Comment:
Yeah good call. I was experimenting with different default values. I'll
tighten up the docs before merging. :)

On Sunday, January 10, 2016, Mitch Robb <notifications@github.com> wrote:

> In source/InfiniteLoader/InfiniteLoader.js
> <https://github.com/bvaughn/react-virtualized/pull/51#discussion_r49288718>
> :
>
> > +    /**
> > +     * Callback to be invoked when more rows must be loaded.
> > +     * It should implement the following signature: ({ startIndex, stopIndex }): Promise
> > +     * The returned Promise should be resolved once row data has finished loading.
> > +     * It will be used to determine when to refresh the list with the newly-loaded data.
> > +     * This callback may be called multiple times in reaction to a single scroll event.
> > +     */
> > +    loadMoreRows: PropTypes.func.isRequired,
> > +    /**
> > +     * Number of rows in list; can be arbitrary high number if actual number is unknown.
> > +     */
> > +    rowsCount: PropTypes.number,
> > +    /**
> > +     * Threshold at which to pre-fetch data.
> > +     * A threshold X means that data will start loading when a user scrolls within X rows.
> > +     * This value defaults to 10.
>
> should be 15, right?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/bvaughn/react-virtualized/pull/51/files#r49288718>.
>